python/src/spark_rapids_ml/umap.py

```diff
@@ -507,19 +526,67 @@ def raw_data(self) -> List[List[float]]:
     def _get_cuml_transform_func(
         self, dataset: DataFrame, category: str = transform_evaluate.transform
     ) -> Tuple[_ConstructFunc, _TransformFunc, Optional[_EvaluateFunc],]:
+        cuml_alg_params = self.cuml_params.copy()
+        driver_embedding = np.array(self.embedding_, dtype=np.float32)
+        driver_raw_data = np.array(self.raw_data_, dtype=np.float32)
+
         def _construct_umap() -> CumlT:
-            raise NotImplementedError("TODO")
+            import cupy as cp
+            from cuml.common import SparseCumlArray, input_to_cuml_array
```

Comment:
please use utils.cudf_to_cuml_array since there is a compatibility issue with the older cuml